docs/sections/versions/baker-3-release-notes.md

```diff
@@ -0,0 +1,215 @@
+This is done using the fireEventAndResolveWhenReceived and fireEventAndResolveWhenCompleted methods.
+
+```scala tab="Scala"
+val result: Future[SensoryEventStatus] = baker.fireEventAndResolveWhenReceived(recipeInstanceId, orderPlaced)
+val result: Future[SensoryEventResult] = baker.fireEventAndResolveWhenCompleted(recipeInstanceId, paymentMade)
+```
+
+```java tab="Java"
+CompletableFuture<SensoryEventStatus> result = baker.fireEventAndResolveWhenReceived(recipeInstanceId, paymentMade);
+CompletableFuture<SensoryEventResult> result = baker.fireEventAndResolveWhenCompleted(recipeInstanceId, firstOrderPlaced);
+```
+
+#### Wait for a specific Event
+A new moment you can ask Baker to notify you on is on a specific Event.
+You can use the fireEventAndResolveOnEvent method for this.
+
+```scala tab="Scala"
+val result: Future[SensoryEventResult] = baker.fireEventAndResolveOnEvent(recipeInstanceId, orderPlaced, "eventName")
+```
+
+```java tab="Java"
+CompletableFuture<SensoryEventResult> result =
+                baker.fireEventAndResolveOnEvent(recipeInstanceId, firstOrderPlaced, "eventName");
+```
+
+#### EventResult
+When firing a event and waiting for completion stage or waiting for a specific event a EventResult object will be returned.
+This EventResult object contains the SensoryEventStatus, event names and created ingredients.
+This can be useful if you need to make a decision depending on the state.
+In these cases there is no need to inquire on Baker itself anymore.
+
+```scala tab="Scala"
+case class SensoryEventResult(
+                               sensoryEventStatus: SensoryEventStatus,
+                               eventNames: Seq[String],
+                               ingredients: Map[String, Value]
+) extends common.SensoryEventResult with ScalaApi
+```
+
+```java tab="Java"
+case class SensoryEventResult(
+                               sensoryEventStatus: SensoryEventStatus,
+                               eventNames: java.util.List[String],
+                               ingredients: java.util.Map[String, Value]
+) extends common.SensoryEventResult with JavaApi {
+
+  def getSensoryEventStatus: SensoryEventStatus = sensoryEventStatus
+
+  def getEventName: java.util.List[String] = eventNames
+
+  def getIngredients: java.util.Map[String, Value] = ingredients
+}
+```
+
+### New EventListener functions
+For the EventListener we have moved away from the annotation based approach.
+In this version you can add a EventListener function instead.
+See the [event listener section](http://localhost:8000/sections/reference/event-listener/) for more details.
+
+## Improved and simplified Runtime
+### Better performance
+Due to simplifying and refactoring the runtime we see a big increase in the load a Baker instance can handle.
+For our created [example project](https://github.com/ing-bank/baker/tree/master/examples/src)
+we saw the project could handle over 5 times the load when switching between version 2 and 3 of Baker.
```

Comment:
"when switching between version 2 and 3 of Baker" => "when switching from version 2 of Baker to version 3" | "when switching from version 2 to 3 of Baker"